apps/voting/app/src/components/VotePanelContent.js

```diff
@@ -73,54 +73,49 @@ class VotePanelContent extends React.Component {
   handleExecuteClick = () => {
     this.props.onExecute(this.props.vote.voteId)
   }
-  loadUserBalance = (user, vote, tokenContract) => {
-    const { tokenDecimals } = this.props
-    if (tokenContract && user) {
-      tokenContract
-        .balanceOfAt(user, vote.data.snapshotBlock)
-        .toPromise()
-        .then(balance => {
-          const adjustedBalance = Math.floor(
-            parseInt(balance, 10) / Math.pow(10, tokenDecimals)
-          )
-          this.setState({ userBalance: adjustedBalance })
-        })
+  loadUserBalance = async (user, vote, tokenContract) => {
+    if (!tokenContract || !user) {
+      return
     }
+
+    const { tokenDecimals } = this.props
+
+    const balance = await tokenContract
```

Comment:
What happens if this fails? I really like `async/await` as a clearer syntax but having to add `catch` statements is not as fun..

Reply:
Yes I agree, `try / catch` are not fun 🙅‍♀️🙅‍♀️🙅‍♀️

I wonder if pattern matching is going to make it less common to throw errors, returning them instead, like [Reason does](https://reasonml.github.io/docs/en/exception) 🤔

But for now `@aragon/api` doesn’t emit errors, so calls never fail.